src/js/modules/infragistics.ui.editors.js

```diff
@@ -8080,6 +8094,44 @@
 
 			this._super();
 		},
+		_validateSpinSettings: function() {
+			var delta = this.options.spinDelta, deltaRanges, index;
+
+			if (typeof delta === "number") {
+				if (delta < 0) {
+					throw new Error(this._getLocaleValue("spinDeltaCouldntBeNegative"));
+				} else {
+					this.options.spinDelta = parseInt(delta, 10);
+				}
+			} else if (typeof delta === "object") {
+				deltaRanges = [ [ "years", 10 ], [ "months", 12 ], [ "days", 28 ],
+					[ "hours", this._dateIndices.hh24 ? 24 : 12 ], [ "minutes", 60 ],
+					[ "seconds", 60 ], [ "milliseconds", 1000 ] ];
+
+				for (index = 0; index < deltaRanges.length; index++) {

```

Comment:
The case is that if spinDelta for years is not defined (it's not in the options), then we need to set it to 1. That's why it's proper to iterate through deltaRanges array. The validation happens on initialization or when set dynamically at run-time, that's why it cannot cause performance issues.